src/segment_ipify.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+type ipify struct {
+	props properties
+	env   environmentInfo
+	IP    string
+}
+
+const (
+	IpifyURL Property = "url"
+	// IpifyCacheTimeout cache timeout
+	IpifyCacheTimeoutout Property = "cache_timeout"
+)
+
+func (i *ipify) enabled() bool {
+	ip, err := i.getResult()
+	if err != nil {
+		return false
+	}
+	i.IP = ip
+
+	return true
+}
+
+func (i *ipify) string() string {
+	segmentTemplate := i.props.getString(SegmentTemplate, "{{.IP}}")
+	template := &textTemplate{
+		Template: segmentTemplate,
+		Context:  i,
+		Env:      i.env,
+	}
+	text, err := template.render()
+	if err != nil {
+		return err.Error()
+	}
+
+	return text
+}
+
+func (i *ipify) getResult() (string, error) {
+	cacheTimeout := i.props.getInt(IpifyCacheTimeoutout, DefaultCacheTimeout)
```

Comment:
this can be the generic `CacheTimeout` property

Reply:
Fixed it in an added commit below.